src/lib/merkle_mask/masking_merkle_tree.ml

```diff
@@ -224,9 +224,34 @@ module Make (Inputs : Inputs_intf.S) = struct
       in
       go locations_with_locations_rev leftover_location_accounts_rev []
 
+    let self_merkle_path t address =
+      Option.try_with (fun () ->
```

Comment:
It seems to me more natural to use `Option.bind` in order to propagate missed lookups rather than throw exceptions.

Reply:
This is targeted in my refactoring for PR #14594 (I'll commit it shortly)